packages/ra-ui-materialui/src/input/AutocompleteArrayInput.spec.tsx

```diff
@@ -812,6 +813,23 @@ describe('<AutocompleteArrayInput />', () => {
         expect(screen.queryByText('New Kid On The Block')).not.toBeNull();
     });
 
+    it('should allow the creation of a new choice by pressing enter', async () => {
+        render(<OnCreate />);
+        const input = (await screen.findByLabelText(
+            'Roles'
+        )) as HTMLInputElement;
+        // Enter an unknown value and submit it with Enter
+        await userEvent.type(input, 'New Value{Enter}');
+        // Clear the input, otherwise the new value won't be shown in the dropdown as it is selected
+        fireEvent.change(input, {
+            target: { value: '' },
+        });
```

Comment:
**suggestion**: It would be nice to have an additional check inbetween those 2 lines, that checks the input value to make sure the new choice is pre-selected.
Manual tests show this is the case, but it would be nice to cover that feature in the unit test too.